imagetagger/imagetagger/images/management/commands/runzipdaemon.py

```diff
@@ -0,0 +1,48 @@
+import fasteners
+import os
+import sys
+from time import sleep
+
+import zipfile
+from django.conf import settings
+from django.core.management import BaseCommand
+from django.db import transaction
+from django.db.models import Q
+
+from imagetagger.images.models import ImageSet
+
+
+class Command(BaseCommand):
+    help = 'Runs a daemon that creates zip archives for image sets and keeps them up to date'
+
+    def handle(self, *args, **options):
+        if not settings.ALLOW_ZIP_DOWNLOAD:
+            sys.stderr.write('To enable zip download, set ALLOW_ZIP_DOWNLOAD to True in settings.py\n')
+            return
+
+        lock = fasteners.InterProcessLock(os.path.join(settings.IMAGE_PATH, 'zip.lock'))
+        gotten = lock.acquire(blocking=False)
+        if gotten:
+            while True:
+                for imageset in ImageSet.objects.filter(~Q(zip_state=ImageSet.ZipState.READY)):
+                    self._regenerate_zip(imageset)
+                sleep(10)
+        else:
+            sys.stderr.write('The lockfile is present. There seems to be another instance of the zip daemon running.'
+                             'Please stop it before starting a new one.\n'
+                             'If this problem persists, delete {}.\n'.format(lock.path))
+
+    def _regenerate_zip(self, imageset):
+        with transaction.atomic():
```

Comment:
why do you use a transaction for this command if you are only doing a single db operation within?